### PR TITLE
[DEV-3004] Using ENV_VARs for API, GA, and MAPBOX

### PR DESCRIPTION
**Pull request by MaxwellKendall**
### ~DNM until Config PR is merged~ merged!

**High level description:**
Configuring webpack to expose env-vars and default values to prevent any breaking changes for users w/o access to these variables.

**Technical details:**
- Separated scss config into dev/prod webpack configs
- Configured webpack to expose env vars
- Used new env vars in `src/js/globalConstants.js`
- Updated readme

**JIRA Ticket:**
[DEV-3004](https://federal-spending-transparency.atlassian.net/browse/DEV-3004)

The following are ALL required for the PR to be merged:
- [x] Code review
- [x] Test in Sandbox
- [x] Config repo updated to set env vars on deploy

### CODE REVIEW

File: webpack/webpack.common.js [hunk 1]
```diff
@@ -10,8 +10,6 @@ const gitRevisionPlugin = new GitRevisionPlugin({ branch: true }); // 'rev-parse
 console.log("Commit Hash for this build: ", gitRevisionPlugin.commithash());
 console.log("Branch for this build: ", gitRevisionPlugin.branch());
 
-const isProduction = (process.env.NODE_ENV === 'production');
```
Comment (MaxwellKendall) [inline webpack/webpack.common.js]: This was always false as `NODE_ENV` isn't set yet.

File: webpack/webpack.common.js [hunk 2]
```diff
@@ -68,20 +66,6 @@ module.exports = {
...
-                    {
-                        loader: "sass-loader",
-                        options: {
-                            sourceMap: !isProduction,
-                            includePaths: ["./src/_scss", "./node_modules"]
-                        }
-                    }
-                ]
-            },
             {
```
Comment (MaxwellKendall) [inline webpack/webpack.common.js]: extracting this to .dev/.production webpack config!

File: tests/testResources/mockGlobalConstants.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+export default {
+    // LOCAL
+    API: "http://localhost:8000/api/",
+    LOCAL_ROOT: "",
+    GITHUB: "",
+    GA_TRACKING_ID: "UA-92617810-1",
```
Comment (ebdabbs) [inline tests/testResources/mockGlobalConstants.js]: Could we use an empty string or a mock id?

File: webpack/webpack.prod.config.js [hunk 4]
```diff
@@ -38,12 +38,44 @@ module.exports = merge(common, {
...
             filename: "[name].[contenthash].css"
         }),
         new webpack.optimize.MinChunkSizePlugin({
             minChunkSize: 300000
+        }),
+        new webpack.DefinePlugin({
+            'process.env': {
+                API: process.env.API
+                    ? JSON.stringify(process.env.API)
+                    : JSON.stringify("http://localhost:8000/api/"),
```
Comment (ebdabbs) [inline webpack/webpack.prod.config.js]: Should this be `https://api.usaspending.gov/` for prod.config?

File: README.md [hunk 5]
```diff
@@ -51,27 +51,9 @@ Assumptions:
...
-
-* `API` (string) is the base API URL for the server that is hosting the API.
-	* The USAspending API is located at `https://api.usaspending.gov/api/`, with the trailing slash included.
-* `DEV` (boolean) indicates if you are running the application in development mode or for production use. Enabling development mode activates certain debugging functionality at the expense of some performance.
-* `PERF_LOG` (boolean) indicates if you want to enable performance logging data in the JavaScript console. This requires `DEV` to be enabled as well.
-
-`DEV` and `PERF_LOG` should be disabled when deploying to a hosted public, staging, or production environment.
-
-**TIP!** You can use separate `GlobalConstants_dev.js` and `GlobalConstants_prod.js` files to point to different API endpoints that align with different environments.
+The `usaspending-website` folder provides a single configuration file named `GlobalConstants.js`. Here you may adjust the `API` property to use either a local api or the production api. By default, `npm run start` uses the local API url. To use the production API, simply overwrite the `API` property or rewrite the `npm` script to use `webpack.prod.config.js`.
```
Comment (ebdabbs) [inline README.md]: Recommend specifying the local API url (http://localhost:8000/api/) in the documentation

**Pull request merged**